extensions/amp-user-notification/0.1/test/test-amp-user-notification.js

```diff
@@ -18,45 +18,37 @@ import {
   AmpUserNotification,
   UserNotificationManager,
 } from '../amp-user-notification';
-import {createIframePromise} from '../../../../testing/iframe';
 import {getServiceForDoc} from '../../../../src/service';
-import * as sinon from 'sinon';
 
 
-describe('amp-user-notification', () => {
+describes.realWin('amp-user-notification', {
+  amp: {
+    ampdoc: 'single',
+    extensions: ['amp-user-notification'],
+  },
+}, env => {
   let sandbox;
-  let iframe;
+  let ampdoc;
+  let win;
   let dftAttrs;
-  let storage;
   let storageMock;
 
   beforeEach(() => {
-    sandbox = sinon.sandbox.create();
+    sandbox = env.sandbox;
+    ampdoc = env.ampdoc;
+    win = env.win;
     dftAttrs = {
       id: 'n1',
       'data-show-if-href': 'https://www.ampproject.org/get/here',
       'data-dismiss-href': 'https://www.ampproject.org/post/here',
       'layout': 'nodisplay',
     };
-  });
-
-  afterEach(() => {
-    if (storageMock) {
```

Comment:
Good catch. Added to all cases.

We can't do in afterEach because for some cases we have to wait a promise to resolve :-(

Also, somehow having assertion in `afterEach` causes Mocha to report incorrect total test number.